polaris/dataset/_subset.py

```diff
@@ -76,6 +76,10 @@ def __init__(
 
         self._adapters = adapters
         self._featurization_fn = featurization_fn
+
+        # NOTE (cwognum): Note to future self. As we're starting to think about competition-style benchmarks,
+        #  we will likely split up datasets. In that case, this default iloc_to_loc mapping won't work.
+        #  By that time, we should probably be able to overwrite this mapping.
```

Comment:
This is a good point, also relevant is the ongoing discussion about making all datasets Zarr-only. There have been a lot of discussions over the years about supporting a dataframe-like interface for Zarr files (e.g. [here](https://github.com/zarr-developers/numcodecs/issues/452), [here](https://github.com/zarr-developers/community/issues/31), and [here](https://github.com/zarr-developers/zarr-python/pull/84)), but as it stands the Zarr team has no intention of supporting dataframe-like access to Zarr files. It will remain array-like only for the time being, which may require re-thinking this kind of indexed access (or maybe not, but just something to investigate more and keep in mind!)

Reply:
Fun fact: `hadim` is an old colleague! His accounts pops up everywhere (at least in most open-source projects relevant to us at Valence). I can actually remember discussing this issue with him! Feels like forever ago :smile:

Reply:
Haha amazing! Small world.